File: 2021/219. Contains Duplicate II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> indexes;
        for (int i = 0; i < nums.size(); i++) {
            if (indexes.find(nums[i]) != indexes.end() && (i - indexes[nums[i]]) <= k) {
                return true;
            }
            indexes[nums[i]] = i;
        }
        return false;
    }
};

File: 2024/219. Contains Duplicate II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        indexes = defaultdict(int)
        for i, n in enumerate(nums):
            if n in indexes and i - indexes[n] <= k:
                return True
            indexes[n] = i
        return False
